tests/e2e.sh

```diff
@@ -127,9 +166,14 @@ echo "access kube-state-metrics metrics endpoint"
 curl -s "http://localhost:8001/api/v1/proxy/namespaces/kube-system/services/kube-state-metrics:8080/metrics" >$KUBE_STATE_METRICS_LOG_DIR/metrics
 
 echo "check metrics format with promtool"
-wget -q -O /tmp/prometheus.tar.gz https://github.com/prometheus/prometheus/releases/download/v$PROMETHEUS_VERSION/prometheus-$PROMETHEUS_VERSION.linux-amd64.tar.gz
-tar zxfv /tmp/prometheus.tar.gz -C /tmp
-cat $KUBE_STATE_METRICS_LOG_DIR/metrics | /tmp/prometheus-$PROMETHEUS_VERSION.linux-amd64/promtool check metrics
+[ -n "$E2E_SETUP_PROMTOOL" ] && setup_promtool
+cat $KUBE_STATE_METRICS_LOG_DIR/metrics | promtool check metrics
+
+collectors="configmap cronjob daemonset deployment endpoint hpa job limitrange namespace node persistentvolumeclaim pod replicaset replicationcontroller resourcequota secret service statefulset"
+for collector in $collectors; do
+    echo "checking that kube_${collector}* metrics exists"
+    grep "^kube_${collector}_" $KUBE_STATE_METRICS_LOG_DIR/metrics
```

Comment:
I feel that if “tests” get any more complicated than this, then we should start implementing them in Go.